split_to_CMYK.jsx

```diff
@@ -2,104 +2,108 @@
-    setHueSatDes1.putReference(c2t("null"), setHueSatRef);
-    setHueSatDes2.putEnumerated(s2t("presetKind"), s2t("presetKindType"), s2t("presetKindCustom"));
-    setHueSatDes3.putEnumerated(s2t("channel"), s2t("channel"), s2t("composite"));
-    setHueSatDes3.putInteger(s2t("hue"), hue);
-    setHueSatDes3.putInteger(c2t("Strt"), sat);
-    setHueSatDes3.putInteger(s2t("lightness"), light);
-    setHueSatList.putObject(s2t("hueSatAdjustmentV2"), setHueSatDes3);
-    setHueSatDes2.putList(s2t("adjustment"), setHueSatList);
-    setHueSatDes1.putObject(s2t("to"), s2t("hueSaturation"), setHueSatDes2);
-    executeAction(s2t("set"), setHueSatDes1, DialogModes.NO);
-    } else {return}
-}
+      setHueSatRef.putEnumerated(s2t("adjustmentLayer"), s2t("ordinal"), s2t("targetEnum"));
+      setHueSatDes1.putReference(c2t("null"), setHueSatRef);
+      setHueSatDes2.putEnumerated(s2t("presetKind"), s2t("presetKindType"), s2t("presetKindCustom"));
+      setHueSatDes3.putEnumerated(s2t("channel"), s2t("channel"), s2t("composite"));
+      setHueSatDes3.putInteger(s2t("hue"), props.h);
+      setHueSatDes3.putInteger(c2t("Strt"), props.s);
+      setHueSatDes3.putInteger(s2t("lightness"), props.v);
+      setHueSatList.putObject(s2t("hueSatAdjustmentV2"), setHueSatDes3);
+      setHueSatDes2.putList(s2t("adjustment"), setHueSatList);
+      setHueSatDes1.putObject(s2t("to"), s2t("hueSaturation"), setHueSatDes2);
+      executeAction(s2t("set"), setHueSatDes1, DialogModes.NO);
+    }; else {return};
+  };
+};
 function runScript() {
-        doc.changeMode(ChangeMode.CMYK);
-    makeLayers();
-    splitChannels('Cyan');
-    splitChannels('Magenta');
-    splitChannels('Yellow');
-    splitChannels('Black');
-        doc.changeMode(ChangeMode.RGB);
-    addLevelsHueSat('Cyan', 174, 50, 20);
-    addLevelsHueSat('Magenta', 300, 50, 20);
-    addLevelsHueSat('Yellow', 50, 50, 20);
-    addLevelsHueSat('Black')
-    	app.refresh();
-    alert("Split to CMYK\n* Use Hue/Saturation Adjustment layers to lighten, darken, change color\n* Use Levels adjustment layer for best contrast\n* Use Layer masks to mask out content\n* Original image is hidden on bottom layer");
+  var CYAN_LAYER = new ChannelLayer('Cyan', BlendMode.NORMAL, 174, 50, 20);
+  var MAGENTA_LAYER = new ChannelLayer('Magenta', BlendMode.MULTIPLY, 300, 50, 20);
+  var YELLOW_LAYER = new ChannelLayer('Yellow', BlendMode.MULTIPLY, 50, 50, 20);
+  var BLACK_LAYER = new ChannelLayer('Black', BlendMode.MULTIPLY, 0, 0, 0);
+  var layers = [CYAN_LAYER, MAGENTA_LAYER, YELLOW_LAYER, BLACK_LAYER];
+
+  doc.changeMode(ChangeMode.CMYK);
+  splitChannels(layers);
+  doc.changeMode(ChangeMode.RGB);
+  createAdjustmentLayers(layers);
+  app.refresh();
+  alert("Split to CMYK\n* Use Hue/Saturation Adjustment layers to lighten, darken, change color\n* Use Levels adjustment layer for best contrast\n* Use Layer masks to mask out content");
 }
 var doc = app.activeDocument;
+doc.duplicate();
+activeDocument = doc;
```

Comment:
This looks like an error too, you're declaring a global var named `activeDocument` instead of bringing the doc to the foreground.  do `doc.activate()` instead.